openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -344,6 +344,15 @@ def search_emails(self, request):
             }
         ]
         """
+        if not request.user.is_staff:
+            return Response(
+                {
+                    'developer_message': 'not_found',
+                    'user_message': 'Not Found'
+                },
+                status=status.HTTP_404_NOT_FOUND
```

Comment:
This should be 403 ... no?

Reply:
I like 404. Better to not acknowledge the presence of the endpoint to non-staff users.

Reply:
Belated review: I'd recommend 403 instead -- 403 means the server understood the request and knows who you are but denies access. If there were PII in the  _route_, e.g. a user lookup endpoint where the response depended on the existence of an email address in the system and the user's specific authorization to see that email address, you might want a 404 instead in order to avoid leaking information, but I don't think that's the case here.